docs/dev/design/new-search-api.rst

```diff
@@ -0,0 +1,238 @@
+             "highlights": {
+               "name": [],
+               "content": [
+                 ", json=data, headers=HEADERS, ) print(response.json()) The content of body.json is like,  &quot;name&quot;: &quot;<span>Test</span>"
+               ]
+             }
+           }
+         ]
+       }
+     ]
+   }
+
+Examples
+--------
+
+- ``project:docs project:dev/latest test``: search for ``test`` in the default
+  version of the ``docs`` project, and in the latest version of the ``dev`` project.
+- ``a project:docs/stable search term``: search for ``a search term`` in the
+  stable version of the ``docs`` project.
+
+- ``project:docs project\:project/version``: search for ``project::project/version`` in the
+  default version of the ``docs`` project.
+
+- ``search``: invalid, at least one project is required.
+
+
+Future features
+---------------
+
+- Allow searching on several versions of the same project
+  (the API response is prepared to support this).
+- Allow specify the type of search:
+
+  - Multi match (query as is)
+  - Simple query string (allows using the ES query syntax)
+  - Fuzzy search (same as multi match, but with with fuzziness)
+
+Questions / pending decisions
+-----------------------------
+
+Integration with the dashboard search.
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The search API and the search from the dashboard
+use the same backend, but they are used in a different way.
+
+The dashboard search by default searches on all projects (.org),
+or all project the current user has access to (.com).
+And the API search searches on explicitly given projects.
+
+The dashboard search allows filtering by version and role,
+the API search allows filtering only by version (and it's required).
+
+The dashboard search makes use of filters in order to return
+the number of results from other versions/roles.
+Is this feature useful? It could slow down the response.
+Searching several versions at the same time could be a better replace?
+
+The dashboard search can be used to search for projects
+by their name and description.
+The API search doesn't support this.
+Is this feature useful? Should we implement
+this as a way to filter projects from https://readthedocs.org/dashboard/ instead?
+This will be using just https://docs.djangoproject.com/en/4.0/ref/contrib/postgres/search/
+or ``contains``.
```

Comment:
I think this is probably a good solution. Having project search just use the DB, and then ES is only for searching files. I'd want to have a different UX even for it though, if it's not using the same backend...